### DISCUSSION TITLE
RISOLTO: Errore "Hai impostato la Cassa professionisti ma non ne hai specificato il tipo"

### DISCUSSION THREAD

**Discussion by Sciabal** [fattureincloud/api#256] [Q&A] [answered]
Ciao,

ho un problema con la validazione formale delle fatture create per procedere alla fattura elettronica.
Tramite API creo correttamente la fattura ma se in FiC faccio una verifica formale mi da l'errore 

_Hai impostato la Cassa professionisti ma non ne hai specificato il tipo_

In realtà sembra tutto corretto, tanto è vero che se apro in modifica la fattura e la salvo subito, senza fare nulla, la verifica formale viene validata.
Premetto che confrontando i campi della fattura elettronica prima (appena creata tramite API) e dopo averla aperta e subito salvata, sono identici.

Da cosa può dipendere?

Grazie!

**Comment by emajo** [comment, marked as answer]
Ciao @Sciabal è necessario specificare il campo `ei_cassa_type`, i valori si possono trovare sul tracciato ufficiale (valori TC01-TC22), oppure direttamente dalla nostra interfaccia web.

Grazie,
Emanuele.

**Reply by Sciabal** [reply]
Ti ringrazio, in effetti mancava quel campo!
Ne approfitto per chiederti se oltre al campo _email_, anche il campo _code_ (codice cliente, per entity) non è aggiungibile con la creazione di una fattura, ma solo lavorando sull'utente in anagrafica.

Grazie!

**Reply by emajo** [reply]
Esatto, il campo code non può venire aggiunto in creazione di fattura come i campi email, phone, fax e intent_*.

**Reply by Sciabal** [reply]
Grazie mille!!
